src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -945,31 +943,28 @@ protected static List<TransitAlert> filterAlerts(
     Collection<TransitAlert> alerts,
     LegacyGraphQLTypes.LegacyGraphQLQueryTypeAlertsArgs args
   ) {
-    var severities = LegacyGraphQLUtils.mapIterableToList(args.getLegacyGraphQLSeverityLevel());
-    var effects = LegacyGraphQLUtils.mapIterableToList(args.getLegacyGraphQLEffect());
-    var causes = LegacyGraphQLUtils.mapIterableToList(args.getLegacyGraphQLCause());
+    var severities = args.getLegacyGraphQLSeverityLevel();
+    var effects = args.getLegacyGraphQLEffect();
+    var causes = args.getLegacyGraphQLCause();
     return alerts
       .stream()
       .filter(alert ->
         args.getLegacyGraphQLFeeds() == null ||
-        ((List<String>) args.getLegacyGraphQLFeeds()).contains(alert.getId().getFeedId())
+        args.getLegacyGraphQLFeeds().contains(alert.getId().getFeedId())
       )
       .filter(alert ->
-        severities.isEmpty() || severities.contains(getLegacyGraphQLSeverity(alert.severity()))
-      )
-      .filter(alert -> effects.isEmpty() || effects.contains(getLegacyGraphQLEffect(alert.effect()))
+        severities == null || severities.contains(getLegacyGraphQLSeverity(alert.severity()))
```

Comment:
Can you please extract a method to convert null to the empty list? It could be part of ListUtils. Then you don't need the repeated null check.

Reply:
But I still have to do repeated isEmpty check (although this can be put into a method like the null check could be put). I'm not sure what is the best solution here. Normally it would be really nice to have the list always as a list so you can use the same logic but here you can't just check if the list contains something if it's null/empty because it would lead to wrong results.